### ISSUE TITLE
[lodash/path-style] false positive with _.matchesProperty

### ISSUE THREAD

**Issue by jrencz** [wix/eslint-plugin-lodash#67]
```json
{
  "rules": {
    "lodash/path-style": ["error", "string"],
  }
}
```
```js
const order = {id: 42};
const belongsToOrder = _.matchesProperty(['order_id', order.id]);
```

It triggers the error from `lodash/path-style`:

<img width="309" alt="screen shot 2016-04-19 at 12 59 04" src="https://cloud.githubusercontent.com/assets/948356/14636937/b318d70e-062e-11e6-858d-97fc4c8d7612.png">

I think this is not the case where this rule should err.
This syntax means "create a function that will return true if the object it takes has a value given as the 2nd element of the tuple on property path given as the first element of the tuple"

So this:

```js
const matches = _.matchesProperty([['foo', 'bar'], value]);
```

should violate the rule with the config given. And this:

```js
const matches = _.matchesProperty(['foo.bar', value]);
```

should be lint-free.

**Comment by ganimomer** [comment]
Looks like you're right. This is probably a remnant from Lodash v3. On it.

**Comment by ganimomer** [comment]
Actually, just checked, and it looks like the code you wrote is wrong.
the **shorthand** requires a tuple as an array, but [the method requires a key-value pair](https://lodash.com/docs#matchesProperty).

For example:
```js
const test = _.matchesProperty(['a','b'])
test({a: 'b'}) // false
test({a: {b: undefined}}) //true
```

So looks like the rule is fine.

**Comment by jrencz** [comment]
@ganimomer I never showed how I use the method. I do pass a key value pair to it, but it doesn't matter how the function `_.matchesProperty` created is used in this case.

```js
const testedObjects = [
  {foo: {bar: 1}},
  {foo: {bar: 2}},
  {foo: {bar: 3}},
]
const matches = _.matchesProperty(['foo.bar', 3]);

const anyMatches = testedObjects.some(matches);
```

**Comment by jrencz** [comment]
Ok, I see the mistake I'm making: `_.matchesProperty` takes `propertyPath` as the first argument and `value` as the second.
 
I passed a `matchesProperty` shorthand into `_.matchesProperty` itself.

**Comment by ganimomer** [comment]
Exactly. What your `matches` example does is check if the object is of the format `{"foo.bar": {3: undefined}}` (or the inner object just doesn't have a `3` property).
It's the same as if you wrote `_.matchesProperty(['foo.bar', 3], undefined)`.

**Comment by jrencz** [comment]
@ganimomer btw: is there a rule that checks if all required params of the lodash method were passed?

**Comment by ganimomer** [comment]
There isn't, and that's because unlike passing *extra* parameters, it's really difficult to analyse if omitting parameters on purpose (which, as far as I know, works in all Lodash methods).
It would just have too many false positives.
For instance, a pattern I see and use a lot is:
```js
_(collection).map(f).find()`
```
Which returns the first truthy result of `f` on any object in `collection`, and thanks to shortcut fusion, halts after the first truthy result it finds.

**Comment by jrencz** [comment]
ok, thanks
